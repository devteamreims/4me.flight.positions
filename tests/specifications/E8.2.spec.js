import request from 'supertest';
import nock from 'nock';
import app from '../../index';

import {getTracker} from 'mock-knex';

describe('spec 8.2', () => {
  const tracker = getTracker();

  beforeEach(() => {
    process.env.POSITION_MAX_AGE = 3600*24*365*20; // 20 years

    tracker.install();
    tracker.on('query', query => {
      query.response([{
        callsign: 'ABC123',
        lat: 49.2534,
        long: 4.033,
        alt: 340,
        when: Math.floor(new Date("Sat, 10 Sep 2016 16:05:11 GMT") / 1000),
      }]);
    })
  });

  afterEach(() => {
    tracker.uninstall();
    delete process.env.POSITION_MAX_AGE;
  });

  test('present SQL data in JSON format and produce a valid output', () => {
    return request(app)
      .get('/?callsigns[]=ABC123')
      .expect('Content-Type', /json/)
      .expect(res => {
        expect(res.body).toMatchSnapshot();
      });
  });
});
