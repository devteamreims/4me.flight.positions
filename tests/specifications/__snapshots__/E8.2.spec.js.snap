exports[`spec 8.2 present SQL data in JSON format and produce a valid output 1`] = `
Array [
  Object {
    "alt": 34000,
    "callsign": "ABC123",
    "lat": 49.2534,
    "long": 4.033,
    "when": 1473523511000
  }
]
`;
